### PR TITLE
Add sync protocol unit tests and fix Sync_Pins.Content equality bug (REM-18b)

### PR DESCRIPTION
**Pull request by yukuku**
## Summary

Implements **REM-18b** from the tech-debt remediation plan: add test coverage for the sync protocol (`Sync_Mabel`, `Sync_Pins`, `Sync_Rp`). Also fixes a latent bug in `Sync_Pins.Content.equals()` surfaced while writing the tests.

## What's new

**101 unit tests** across 4 new test classes in `Alkitab/src/test/java/yuku/alkitab/base/sync/`:

- **`SyncDeltaTest.kt`** (34 tests) — `SyncAdapter.patchNoConflict` with add/mod/del, missing GIDs, same-gid-different-kind, multiple mods (last-wins), add-then-del / del-then-add, mixed operations, input-list non-mutation; plus `Sync.entitiesEqual`, `SyncUtils.findEntity`, `SyncUtils.isSameContent`, and round-trip delta-compute-then-apply tests covering additive, modification-only, deletion-only, mixed, and conflict scenarios.
- **`Sync_MabelTest.kt`** (28 tests) — `updateMarker/Label/Marker_LabelWithEntityContent` (covering `_id` preservation when server sends a `mod` for an existing row, all three `Marker.Kind` values, and mutate-in-place semantics) plus `Content` equals/hashCode/toString (long captions, newline escaping, null fields, gid truncation).
- **`Sync_PinsTest.kt`** (23 tests) — `Pin` and `Content` equals/hashCode/toString, including the order-insensitive case now that the bug is fixed.
- **`Sync_RpTest.kt`** (16 tests) — reading-plan `Content` equals/hashCode/toString, including ""started with zero completions"" and null-field edge cases.

**Test infrastructure:** added `Alkitab/src/test/java/android/util/Pair.java` — a test-only replacement for `android.util.Pair`. The Android mockable jar throws `RuntimeException(""Method not mocked"")` on `Pair.create`, which blocked testing `SyncAdapter.patchNoConflict`. This stub mirrors the real class's behavior and is scoped to the unit-test classpath only.

## Bug fix: `Sync_Pins.Content.equals()`

The method copies `pins` into `list1`/`list2`, sorts them by `preset_id`, then never uses the sorted copies — it compared the original unsorted lists. This defeated the intended order-insensitive equality.

```diff
-				if (!pins.equals(content.pins)) return false;
+				if (!list1.equals(list2)) return false;
```

In practice this was masked because `getEntitiesFromCurrent()` always iterates `preset_id` in ascending order, so local-built `Content`s were already sorted identically. But a deserialized sync shadow with pins in a different order would incorrectly compare unequal and trigger a spurious `mod` sync op. The new `content_equals_samePinsDifferentOrder_true` test verifies the fix.

## Test plan

- [x] `./gradlew :Alkitab:testPlainDebugUnitTest --tests ""yuku.alkitab.base.sync.*""` — 101/101 pass
- [x] `./gradlew :Alkitab:testPlainReleaseUnitTest --tests ""yuku.alkitab.base.sync.*""` — 101/101 pass
- [x] No changes to production sync code beyond the one-line `Sync_Pins.Content.equals` fix
- [x] No new runtime dependencies (test `Pair.java` is test-scope only)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Alkitab/src/main/java/yuku/alkitab/base/sync/Sync_Pins.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ public boolean equals(final Object o) {
 				Collections.sort(list1, listSorter);
 				Collections.sort(list2, listSorter);
 
-				if (!pins.equals(content.pins)) return false;
+				if (!list1.equals(list2)) return false;
```
[the remark this answers is not in the document]
Comment (yukuku) [reply Alkitab/src/main/java/yuku/alkitab/base/sync/Sync_Pins.java:138]: Good catch — fixed in 129d08de. `hashCode` now sorts a copy of `pins` by the same `listSorter` before hashing, so it matches the order-insensitive behavior of `equals` and restores the Object contract.

File: Alkitab/src/test/java/yuku/alkitab/base/sync/Sync_PinsTest.kt [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    @Test
+    fun content_equals_samePinsDifferentOrder_true() {
+        // Content.equals is intentionally order-insensitive: it sorts both pin lists
+        // by preset_id before comparing. This matters when a deserialized shadow
+        // happens to list pins in a different order than getEntitiesFromCurrent() —
+        // an order-sensitive comparison would spuriously flag "mod".
+        val a = content(pin(0, 10), pin(1, 20), pin(2, 30))
+        val b = content(pin(2, 30), pin(0, 10), pin(1, 20))
+        assertEquals(a, b)
```
[the remark this answers is not in the document]
Comment (yukuku) [reply Alkitab/src/test/java/yuku/alkitab/base/sync/Sync_PinsTest.kt:140]: Agreed — strengthened in 129d08de. The `content_equals_samePinsDifferentOrder_true` test now also asserts `a.hashCode() == b.hashCode()`, which would have surfaced the hashCode contract violation.
